exporter/elasticsearchexporter/internal/serializer/otelserializer/profile.go

```diff
@@ -0,0 +1,86 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package otelserializer // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/elasticsearchexporter/internal/serializer/otelserializer"
+
+import (
+	"bytes"
+	"encoding/json"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pprofile"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/elasticsearchexporter/internal/serializer/otelserializer/serializeprofiles"
+)
+
+const (
+	stackTraceIndex  = "profiling-stacktraces"
+	stackFrameIndex  = "profiling-stackframes"
+	executablesIndex = "profiling-executables"
+)
+
+// SerializeProfile serializes a profile into the specified buffer
+func SerializeProfile(resource pcommon.Resource, scope pcommon.InstrumentationScope, profile pprofile.Profile, callback func(*bytes.Buffer, string, string) error) error {
```

Comment:
`callback` is very general. We know that the purpose is to push data into the bulk indexer. Can we rename it to, for example `pushData`?